src/com/facebook/buck/swift/SwiftCompile.java

```diff
@@ -90,6 +92,7 @@ public String apply(String input) {
       };
 
   SwiftCompile(
+      CxxLink delegateLinkRule,
```

Comment:
We take the output of the compilation and pass it as an input to the linking in cxx, is there a particular reason for this pattern of delegating the build steps here?